### PR TITLE
Fix audit issues

### PR DESCRIPTION
**Pull request by sn0wba**
- [x] APP-01
- [x] PAR-01
- [x] PAR-02
- [x] TYP-01
- [x] 6D4-01
- [x] HOK-01

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -490,7 +490,6 @@ func NewGotabitApp(
 		appCodec, keys[minttypes.StoreKey], app.GetSubspace(minttypes.ModuleName),
 		app.AccountKeeper, app.BankKeeper, app.DistrKeeper, app.EpochsKeeper, authtypes.FeeCollectorName,
 	)
-	app.MintKeeper.SetHooks(minttypes.MultiMintHooks{})
```
Comment (jacksoom) [inline app/app.go:493]: cause panic elsewhere？

File: x/mint/types/params.go [hunk 2]
```diff
@@ -161,7 +161,7 @@ func validateReductionPeriodInEpochs(i interface{}) error {
 	}
 
 	if v <= 0 {
-		return fmt.Errorf("max validators must be positive: %d", v)
+		return fmt.Errorf("ReductionPeriodInEpochs must be positive: %d", v)
```
Comment (jacksoom) [inline x/mint/types/params.go:164]: reduction factor must be positive

File: x/mint/types/params.go [hunk 3]
```diff
@@ -173,12 +173,12 @@ func validateReductionFactor(i interface{}) error {
...
 	}
 
-	if v.GT(sdk.NewDec(1)) {
+	if v.GTE(sdk.NewDec(1)) {
 		return fmt.Errorf("reduction factor cannot be greater than 1")
 	}
 
-	if v.IsNegative() {
-		return fmt.Errorf("reduction factor cannot be negative")
+	if v.LTE(sdk.NewDec(0)) {
```
Comment (jacksoom) [inline x/mint/types/params.go:180]: Zero is OK right here.

File: app/app.go (hunk 1 above)
Comment (sn0wba) [reply app/app.go:493]: I'm not sure

File: x/mint/types/params.go (hunk 2 above)
Comment (sn0wba) [reply x/mint/types/params.go:164]: There's another function called `validateReductionFactor`
Both are "reduction factor"?

File: x/mint/types/params.go [hunk 4]
```diff
@@ -173,12 +173,12 @@ func validateReductionFactor(i interface{}) error {
 		return fmt.Errorf("invalid parameter type: %T", i)
 	}
 
-	if v.GT(sdk.NewDec(1)) {
+	if v.GTE(sdk.NewDec(1)) {
```
[the remark this answers is not in the document]
Comment (sn0wba) [reply x/mint/types/params.go:176]: The audit report says reduction factor must be less than 1.

File: x/mint/types/params.go (hunk 3 above)
Comment (sn0wba) [reply x/mint/types/params.go:180]: Ok, got it

File: x/mint/types/params.go (hunk 2 above)
Comment (jacksoom) [reply x/mint/types/params.go:164]: reduction period

File: app/app.go (hunk 1 above)
Comment (jacksoom) [reply app/app.go:493]: > I'm not sure

Pls ensure not to cause panic elsewhere

File: x/mint/types/params.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (jacksoom) [reply x/mint/types/params.go:176]: redundant

**Pull request closed without merging**